app/src/androidTest/java/com/github/sdp_begreen/begreen/fragments/CameraFragmentTest.kt

```diff
@@ -1,41 +1,188 @@
+        // shell command, which is a trick that doesn't always work. So we will test only the granted path.
+
+        launchFragmentInContainer { fragment }
+
+        // Click the add new post button
+        onView(withId(R.id.addNewPostBtn)).perform(ViewActions.click())
+
+        // Check if the camera intent is opened
+        Intents.intended(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE))
+    }
+
+    @Test
+    fun startShareActivityAfterTakingPhotoWithCameraAndGettingResultOK() {
+
+        launchFragmentInContainer { fragment }
+
+        // Define a correct result that the camera should return
+        val cameraResult = Instrumentation.ActivityResult(Activity.RESULT_OK, correctCameraResponseIntent)
+
+        // We expect that the camera activity will return "result"
+        Intents.intending(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE)).respondWith(cameraResult)
+
+        // Click the add new post button to start the camera Activity
+        onView(withId(R.id.addNewPostBtn)).perform(ViewActions.click())
+
+        // Verify that the camera activity was actually launched
+        Intents.intended(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE))
+
+        // Verify that the SharePostActivity has been launched
+        Intents.intended(
+            Matchers.allOf(
+                IntentMatchers.hasComponent(SharePostActivity::class.java.name),
+                IntentMatchers.hasExtra(CameraFragment.EXTRA_IMAGE_BITMAP, image)
+            )
+        )
+    }
+
     @Test
-    fun testCameraFragment() {
-        launchFragmentInContainer<CameraFragment>()
-        onView(withId(R.id.cameraFragmentTextView)).check(matches(withText("Fragment where we can display the camera")))
+    fun doNotStartShareActivityAfterCancelingPhotoTakenWithCamera() {
+
+        launchFragmentInContainer { fragment }
+
+        // Define a correct result that the camera should return
+        val cameraResult = Instrumentation.ActivityResult(Activity.RESULT_CANCELED, correctCameraResponseIntent)
+
+        // We expect that the camera activity will return "result"
+        Intents.intending(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE)).respondWith(cameraResult)
+
+        // Click the add new post button to start the camera Activity
+        onView(withId(R.id.addNewPostBtn)).perform(ViewActions.click())
+
+        // Verify that the camera activity was actually launched
+        Intents.intended(IntentMatchers.hasAction(MediaStore.ACTION_IMAGE_CAPTURE))
+
+        // Check that we resume the previous activity.
+        // When debbuging, we noticed that for this test the value of state was RESUMED, while the state
+        // of the test startShareActivityAfterTakingPhotoWithCameraAndGettingResultOK was CREATED. The state seems to be independent
+        // of the time since we tested at different instants with thread.sleep.
```

Comment:
It might be helpful to add additional comments to explain why we care about the state being independent of time and how it affects the test.

Reply:
Sorry if the comments weren't clear. Actually it was not necessary to mention the time but since I did it I will explain it a bit more.